### PR TITLE
chore: simple on-failure mechanism for alert spec

### PR DESCRIPTION
**Pull request by talboren**

### CODE REVIEW

File: docs/platform/core/syntax/basic-syntax.mdx [hunk 1]
```diff
@@ -118,4 +119,14 @@ actions:
...
 
-_*** Actions are the last section of an alert_
\ No newline at end of file
+### On-failure
+```yaml
+on-failure:
+    # Just need a provider we can use to send the failure reason
+    provider:
+      type: slack
+      config: " {{ providers.slack-demo }} "
```
Comment (shahargl) [inline docs/platform/core/syntax/basic-syntax.mdx:128]: add the mesage ("Alert failed to run")

File: docs/platform/core/syntax/basic-syntax.mdx [hunk 2]
```diff
@@ -51,6 +51,7 @@ alert:
 - Metadata (id, description. owners and tags will be added soon)
 - `steps` - list of steps
 - `actions` - list of actions
+- `on-failure` - a conditionless action used in case of an alert failure
```
Comment (shahargl) [inline docs/platform/core/syntax/basic-syntax.mdx:54]: I would remove the "-" and keep it onfailure

**Review by shahargl — APPROVED** [review]
lgtm

File: docs/platform/core/syntax/basic-syntax.mdx (hunk 1 above)
Comment (talboren) [reply docs/platform/core/syntax/basic-syntax.mdx:128]: It's being created when the exception occurs. I'll refactor this soon so merging for now.

**Pull request merged**